cmd/fs-v1-multipart.go

```diff
@@ -60,13 +60,18 @@ func (fs fsObjects) deleteUploadsJSON(bucket, object, uploadID string) error {
 	return fsRemoveMeta(multipartBucketPath, uploadsMetaPath, tmpDir)
 }
 
+func (fs fsObjects) deleteObjectUploadsDir(bucket, object string) error {
+	basePath := pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket)
+	return fsDeleteFile(basePath, pathJoin(basePath, object))
+}
+
 // Removes the uploadID, called either by CompleteMultipart of AbortMultipart. If the resuling uploads
 // slice is empty then we remove/purge the file.
-func (fs fsObjects) removeUploadID(bucket, object, uploadID string, rwlk *lock.LockedFile) error {
+func (fs fsObjects) removeUploadID(bucket, object, uploadID string, rwlk *lock.LockedFile) (bool, error) {
```

Comment:
I think this method could directly use logic in `deleteObjectUploadsDir()` in `defer` lilke 
```go
defer func() {
  if (err == nil) {
    basePath := pathJoin(fs.fsPath, minioMetaMultipartBucket, bucket)
    derr := fsDeleteFile(basePath, pathJoin(basePath, object))
    errorIf(derr, "unable to remove %s in %s", pathJoin(basePath, object), basePath)
  }
}
```

Reply:
addressed